File: basics/oops/1_greeting.py
"""This program is about greeting a student.
"""

import pprint as pp


class Student(object):
    def __init__(self, student_id, student_age, first_name, last_name) -> None:
        self.student_id: int = student_id
        self.student_age: int = student_age
        self.first_name: str = first_name
        self.last_name: str = last_name
        self.student_name: str = first_name + last_name

    def greeting(self) -> None:
        pp.pprint(f"Hello, {self.student_name}")

    def get_age(self) -> None:
        pp.pprint(f"Your age is --> {self.student_age}")


def main():
    my_obj = Student(1, 20, "Yash", "Sonune")
    my_obj.greeting()
    my_obj.get_age()
    return


if __name__ == "__main__":
    main()

File: basics/oops/1_greeting.cpp
#include <bits/stdc++.h>

using namespace std;

class Student {

  int student_id;
  string student_name;
  int student_age;
  string first_name;
  string last_name;

  Student(int id, int age, string fname, string lname) {
    student_id = id;
    student_age = age;
    first_name = fname;
    last_name = lname;
    student_name = first_name + last_name;
    return;
  }

  ~Student() {
    int id = student_id;
    int age = student_age;
    string full_nmae = first_name + " " + last_name;

    cout << "Freed memory for --> " << student_id << endl
         << student_name << endl
         << student_age << endl;
  }
};

void print_name(Student *&my_obj) {
  cout << "Hello, " << my_obj->student_name << endl;
}

void print_age(Student *&my_obj) {
  cout << "Your age is " << my_obj->student_age << endl;
}

int main(int argc, char *argv[]) {
  Student *my_obj = new Student(1, 20, "Yash", "Sonune");
  return 0;
}
